tests/components/sensor/test_mqtt.py

```diff
@@ -42,6 +46,47 @@ def test_setting_sensor_value_via_mqtt_message(self):
         self.assertEqual('fav unit',
                          state.attributes.get('unit_of_measurement'))
 
+    def test_setting_sensor_value_expires(self):
+        """Test the expiration of the value."""
+        mock_component(self.hass, 'mqtt')
+        assert setup_component(self.hass, sensor.DOMAIN, {
+            sensor.DOMAIN: {
+                'platform': 'mqtt',
+                'name': 'test',
+                'state_topic': 'test-topic',
+                'unit_of_measurement': 'fav unit',
+                'expire_after': '2',
+                'force_update': True
+            }
+        })
+
+        time = datetime(2017, 1, 1, 1, tzinfo=dt_util.UTC)
+        fire_time_changed(self.hass, time)
+
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('unknown', state.state)
+
+        fire_mqtt_message(self.hass, 'test-topic', '100')
+        self.hass.block_till_done()
+
+        state = self.hass.states.get('sensor.test')
+        self.assertEqual('100', state.state)
+
+        time = time + timedelta(seconds=1)
+        fire_time_changed(self.hass, time)
+
+        """ Not yet expired """
```

Comment:
Please prefix comments with `# `. Only doc strings (first comment in a function) should contain triple quotes.